File: concrete/src/test/java/pcb/uwu/amount/composite/finance/RentTest.java
package pcb.uwu.amount.composite.finance;

import org.junit.Test;
import pcb.uwu.amount.quantity.Money;
import pcb.uwu.amount.quantity.Time;
import pcb.uwu.units.composite.finance.InterestRateUnit;

import java.util.Currency;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;
import static pcb.uwu.unit.quantity.DayUnit.DAY;
import static pcb.uwu.unit.quantity.WeekUnit.WEEK;

public class RentTest {
	private Money money = new Money(6, Currency.getInstance("GBP"));
	private Time time = new Time(3, DAY);
	private InterestRate interestRate = new InterestRate(0.2, new InterestRateUnit(WEEK));

	@Test
	public void testFromMoneyAndTime() {
		assertEquals("2.00£/d", money.dividedBy(time, DECIMAL64).toString());
		assertEquals("1.200£/wk", money.multipliedBy(interestRate, DECIMAL64).toString());

		System.err.println(interestRate.getClass().getComponentType());
	}
}

File: concrete/src/test/java/pcb/uwu/amount/base/currency/RentTest.kt
package pcb.uwu.amount.base.currency

import pcb.uwu.amount.base.Time
import pcb.uwu.amount.derived.finance.InterestRate
import pcb.uwu.unit.base.DayUnit
import pcb.uwu.unit.base.WeekUnit
import pcb.uwu.unit.derived.fundamental.FrequencyUnit
import kotlin.test.Test
import kotlin.test.assertEquals

class RentTest
{
    @Test
    fun testFromMoneyAndTime()
    {
        assertEquals(expected = "2 £⋅d⁻¹",
                     actual = (GBP(6) / Time(amount = 3, unit = DayUnit)).toString())
        assertEquals(expected = "1.2 £⋅wk⁻¹",
                     actual = (GBP(6) * InterestRate(amount = 0.2, unit = FrequencyUnit(WeekUnit))).toString())
    }
}